src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java

```diff
@@ -657,18 +657,43 @@ protected Model createCleanPom( Model effectivePom ) throws MojoExecutionExcepti
         }
 
         // transform profiles...
+        Dependencies managedDependencies = new Dependencies();
+        if ( effectivePom.getDependencyManagement() != null && effectivePom.getDependencyManagement().getDependencies() != null )
+        {
+            managedDependencies.addAll( effectivePom.getDependencyManagement().getDependencies() );
+        }
+
         for ( Profile profile : effectivePom.getProfiles() )
         {
             if ( !isEmbedBuildProfileDependencies() || !isBuildTimeDriven( profile.getActivation() ) )
             {
                 if ( !isEmpty( profile.getDependencies() ) || !isEmpty( profile.getRepositories() ) )
                 {
-                    Profile strippedProfile = new Profile();
-                    strippedProfile.setId( profile.getId() );
-                    strippedProfile.setActivation( profile.getActivation() );
-                    strippedProfile.setDependencies( profile.getDependencies() );
-                    strippedProfile.setRepositories( profile.getRepositories() );
-                    cleanPom.addProfile( strippedProfile );
+                    List<Dependency> strippedDependencies = new ArrayList<>();
+                    for ( Dependency dep : profile.getDependencies() )
+                    {
+                        Dependency parsedDep = dep.clone();
+                        if ( managedDependencies.contains( parsedDep ) )
+                        {
+                            parsedDep.setVersion( managedDependencies.resolve( parsedDep ).getVersion() );
+                            parsedDep.setScope( managedDependencies.resolve( parsedDep ).getScope() );
```

Comment:
check again to see if `managedDependencies` already has <scope> and <optional> parsed in -- if not, we need to parse them into the flattened pom as well where <scope> is compile if not specified and <optional> is false if not specified.

Reply:
I have set the \<scope\> as compile and \<optional\> as false if they are not specified.

Reply:
LGTM